elasticsearch/src/main/java/org/hibernate/search/elasticsearch/schema/impl/DefaultElasticsearchSchemaTranslator.java

```diff
@@ -151,13 +152,11 @@ private void addPropertyMapping(ElasticsearchMappingBuilder mappingBuilder, Docu
 		// Create facet fields if needed: if the facet has the same name as the field, we don't need to create an
 		// extra field for it
 		for ( FacetMetadata facetMetadata : fieldMetadata.getFacetMetadata() ) {
-			if ( !facetMetadata.getAbsoluteName().equals( fieldMetadata.getAbsoluteName() ) ) {
-				try {
-					addPropertyMapping( mappingBuilder, facetMetadata );
-				}
-				catch (IncompleteDataException e) {
-					LOG.debug( "Not adding a mapping for facet " + facetMetadata.getAbsoluteName() + " because of incomplete data", e );
-				}
+			try {
+				addFieldMapping( propertyMapping, mappingBuilder, facetMetadata );
+			}
+			catch (IncompleteDataException e) {
+				LOG.debug( "Not adding a mapping for facet " + facetMetadata.getAbsoluteName() + " because of incomplete data", e );
```

Comment:
That code goes back to long before this PR (it's even been refactored at least twice), and similar debug messages are generated for other types of mappings (regular fields, bridge-defined fields). Probably not something to address in this PR.
I created a separate JIRA: https://hibernate.atlassian.net/browse/HSEARCH-2458